File: examples/typescript/createJson.ts
import * as fs from 'fs';
import * as path from 'path';

// Get the current directory
const CURRENT_DIR = path.dirname(__filename);

// Load JSON files
const assets = JSON.parse(fs.readFileSync(path.join(CURRENT_DIR,  '..', '..', 'assets.json'), 'utf-8'));
const chains = JSON.parse(fs.readFileSync(path.join(CURRENT_DIR,  '..', '..', 'chains.json'), 'utf-8'));
const cdn = JSON.parse(fs.readFileSync(path.join(CURRENT_DIR,  '..', '..', 'cdn.json'), 'utf-8'));

let data: any[] = [];

for (const asset of assets) {
    // Deepcopy the asset
    const finalAsset = JSON.parse(JSON.stringify(asset));

    // Update the icon to use the proper full cdn path
    finalAsset['icon'] = cdn['icon'] + finalAsset['icon'];

    for (const network of finalAsset["networks"]) {
        // Convert chainPath to chainId
        const [majorNetwork, chain] = network['chainPath'].split('.');
        network["chainId"] = chains[majorNetwork][chain];

        // Remove chainPath
        delete network['chainPath'];
    }

    data.push(finalAsset);
}

// Log the resulting JSON to the console
console.log(JSON.stringify(data, null, 4));

File: examples/python/create_json.py
from copy import deepcopy
import json
import os


CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))

with open(os.path.join(CURRENT_DIR, '..', '..', 'assets.json'), 'r') as infile:
	assets = json.load(infile)

with open(os.path.join(CURRENT_DIR, '..', '..', 'chains.json'), 'r') as infile:
	chains = json.load(infile)

with open(os.path.join(CURRENT_DIR, '..', '..', 'cdn.json'), 'r') as infile:
	cdn = json.load(infile)

data = []

for asset in assets:
	final_asset = deepcopy(asset)

	# Update the icon to use the proper full cdn path
	final_asset['icon'] = cdn['icon'] + final_asset['icon']

	for network in final_asset["networks"]:

		# Convert chainPath to chainId
		major_network, chain = network['chainPath'].split('.')
		network["chainId"] = chains[major_network][chain]
		
		# And remove chainPath
		del network['chainPath']

	data.append(final_asset)

print(json.dumps(data, indent = 4))
